### PR TITLE
chore: Update Atlas SDK to v20230201008

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

Update Atlas SDK to latest version and add action to update.

_Jira ticket:_ https://jira.mongodb.org/browse/INTMDB-1120

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

## Further comments

Screenshot 2023-09-18 at 15 19 04

### CODE REVIEW

File: cfn-resources/api-key/test/apikey.create-sample-cfn-request.sh [hunk 1]
```diff
@@ -11,7 +25,7 @@ set -o pipefail
 profile="default"
 orgId="${MONGODB_ATLAS_ORG_ID}"
 
-projectName="cfn-bot-apikey-test"
```
Comment (maastha) [inline cfn-resources/api-key/test/apikey.create-sample-cfn-request.sh:14]: `cfn-bot-apikey-test` secret was already existing in AWS account and can't be deleted for 7 days (AWS policy on secret deletion), hence had to create a new one.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -48,6 +48,22 @@ Examples:
   - If the PR has `BREAKING CHANGE`: in its description is a breaking change
 - `remove!: description of the PR`: The commit removes a feature from the product. Typically features are deprecated first for a period of time before being removed. Removing a feature is a breaking change (correlating with MAJOR in Semantic Versioning).
 
+## Discovering New API features
```
Comment (AgustinBettati) [inline CONTRIBUTING.md:51]: Can we additionally add the steps for updating a major version? https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md#sdk-major-release-update-procedure

Comment (maastha) [reply CONTRIBUTING.md:51]: updated

**Comment by maastha** [comment]
`api-key `test failure is transient and can be ignored for now.

File: cfn-resources/api-key/cmd/resource/resource.go [hunk 3]
```diff
@@ -168,7 +169,7 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *peErr, nil
 	}
 	// Set the roles from model
-	apiKeyInput := atlasSDK.CreateAtlasOrganizationApiKey{
+	apiKeyInput := atlasSDK.UpdateAtlasOrganizationApiKey{
```
Comment (marcosuma) [inline cfn-resources/api-key/cmd/resource/resource.go:172]: do you know why we were using the `create..` here?

**Review by marcosuma — APPROVED** [review]
only minor comment, LGTM

**Review by AgustinBettati — APPROVED** [review]

File: cfn-resources/api-key/cmd/resource/resource.go (hunk 3 above)
Comment (maastha) [reply cfn-resources/api-key/cmd/resource/resource.go:172]: This was a change [released](https://github.com/mongodb/atlas-sdk-go/releases/tag/v20230201008.0.0) in the latest version of the SDK
`(*ProgrammaticAPIKeysApiService).UpdateApiKey: changed from func(context.Context, string, string, *CreateAtlasOrganizationApiKey) UpdateApiKeyApiRequest to func(context.Context, string, string, *UpdateAtlasOrganizationApiKey) UpdateApiKeyApiRequest`

Note that only the input data type was changes but the API call is correct and still made using `.UpdateApiKey`

**Pull request merged**